client-js/queryEditor/QueryEditor.js

```diff
@@ -118,6 +119,15 @@ class QueryEditor extends React.Component {
     })
   }
 
+  formatQuery = () => {
+    const { query } = this.state
+    const result = sqlFormatter.format(query.queryText)
+    query.queryText = result
+    this.setState({
+      queryText: result
```

Comment:
Can you change setState to:

```js
this.setState({ 
  query: query 
});
```

I think what you have appears to work because javascript objects are passed by reference, so `query.queryText = result` changes the state data, and the this.setState() call forces a redraw of the editor, which picks up on the assignment in line 125. 

`this.setState({ queryText })` sets `this.state.queryText`, which isn't used anywhere. (SqlEditor component references `this.state.query.queryText`)